ganttproject/src/test/java/net/sourceforge/ganttproject/storage/ProjectDatabaseTest.kt

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2020 BarD Software s.r.o, Anastasiia Postnikova
+
+This file is part of GanttProject, an opensource project management tool.
+
+GanttProject is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+GanttProject is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GanttProject.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package net.sourceforge.ganttproject.storage
+
+import net.sourceforge.ganttproject.GanttTask
+import net.sourceforge.ganttproject.storage.tables.Task.*
+import org.easymock.EasyMock.*
+import org.h2.jdbcx.JdbcDataSource
+import org.jooq.SQLDialect
+import org.jooq.impl.DSL
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.awt.Color
+import javax.sql.DataSource
+
+class ProjectDatabaseTest {
+  private lateinit var dataSource: DataSource
+  private lateinit var projectDatabase: ProjectDatabase
+
+  @BeforeEach
+  private fun init() {
+    val ds = JdbcDataSource()
+    ds.setURL(H2_IN_MEMORY_URL)
+    dataSource = ds
+    projectDatabase = SqlProjectDatabaseImpl(dataSource)
+  }
+
+  @AfterEach
+  private fun clear() {
+    dataSource.connection.use { conn ->
+      conn.createStatement().execute("shutdown")
+    }
+  }
+
+  @Test
+  fun `test init creates tables`() {
+    projectDatabase.init()
+    val tasks = DSL.using(dataSource, SQLDialect.H2)
+        .selectFrom(TASK)
+        .fetch()
+    assert(tasks.isEmpty())
+  }
+
+  @Test
+  fun `test insert task`() {
+    projectDatabase.init()
+
+    val mockTask: GanttTask = niceMock(GanttTask::class.java)
```

Comment:
in fact, you can use the real tasks instead of mocks. There is a useful `TestSetupHelper` class which can create an instance of TaskManager for you

Reply:
That's cool!
Seems like the test file should be moved to `ganttproject-tester`, right?